src/assets/scripts/client/aircraft/AircraftCollection.js

```diff
@@ -0,0 +1,213 @@
+ * This collection also keeps a list of `AircraftDefinitionModel` objects, which define each
+ * aircraft type.
+ *
+ * @class AircraftCollection
+ * @extends BaseCollection
+ */
+/* istanbul ignore next */
+export default class AircraftCollection extends BaseCollection {
+    /**
+     * @constructor
+     * @for AircraftCollection
+     * @param aircraftDefinitionList {array<object>}
+     * @param airlineCollection {AirlineCollection}
+     * @param navigationLibrary {NavigationLibrary}
+     */
+    constructor(aircraftDefinitionList, airlineCollection, navigationLibrary) {
+        super(aircraftDefinitionList, airlineCollection, navigationLibrary);
+
+        if (!_isArray(aircraftDefinitionList) || _isEmpty(aircraftDefinitionList)) {
+            // eslint-disable-next-line max-len
+            throw new TypeError('Invalid aircraftDefinitionList passed to AircraftCollection. Expected and array but ' +
+                `received ${typeof aircraftDefinitionList}`);
+        }
+
+        // TODO: this may need to use instanceof instead, but that may be overly defensive
+        if (!_isObject(airlineCollection) || !_isObject(navigationLibrary)) {
+            throw new TypeError('Invalid parameters. Expected airlineCollection and navigationLibrary to be defined');
+        }
+
+        this._airlineCollection = airlineCollection;
+        this._navigationLibrary = navigationLibrary;
+        this.definitionList = [];
+
+        this.init(aircraftDefinitionList);
+    }
+
+    /**
+     * Lifecycle method. Should be run only once on instantiation.
+     *
+     * Initializes class properties.
+     *
+     * @for AircraftCollection
+     * @method init
+     * @param aircraftDefinitionList {array<object>}
+     */
+    init(aircraftDefinitionList) {
+        this.definitionList = this._buildAircraftDefinitionList(aircraftDefinitionList);
+    }
+
+    /**
+     * Callback method fired by an interval defined in the `SpawnScheduler`.
+     *
+     * This is the entry method for creating new departing and arriving aircraft.
+     * This method should only be called as a callback from a `SpawnScheduler` timer.
+     *
+     * @for AircraftCollection
+     * @method createAircraftWithSpawnModel
+     * @param spawnModel {SpawnPatternModel}
+     */
+    createAircraftWithSpawnModel = (spawnModel) => {
+        const initializationProps = this._buildAircraftProps(spawnModel);
+        const aircraftModel = new AircraftInstanceModel(initializationProps);
+
+        console.log('SPAWN :::', spawnModel.category, initializationProps);
+
+        this.addItem(aircraftModel);
+    };
+
+    /**
+     * Add an `AircraftInstanceModel` to the collection
+     *
+     * @for AircraftCollection
+     * @method addItem
+     * @param aircraftModel
+     */
+    addItem(aircraftModel) {
+        // TODO: add instanceof check here
+        this._items.push(aircraftModel);
+        // FIXME: TOTAL HACK!!! REMOVE ME
+        prop.aircraft.list.push(aircraftModel);
```

Comment:
Yeah haha I was going to ask and I figured you had plans to deal with it, hence the fixme. ;)